scrape/manager.go

```diff
@@ -100,6 +100,40 @@ func (mc *MetadataMetricsCollector) Collect(ch chan<- prometheus.Metric) {
 	}
 }
 
+// ReadyManager is an interface to represent a ReadyScrapeManager.
+type ReadyManager interface {
```

Comment:
Don't we usually put these in main?

This is nothing to do with scraping, it's purely for main's sake.

Reply:
They used to be close to the source, at some point they were moved to main. Great call 👍  I struggled to find a home for this.